tests-e2e/appstudio/webhook_test.go

```diff
@@ -220,18 +220,66 @@ var _ = Describe("Webhook E2E tests", func() {
 			Expect(err).NotTo(Succeed())
 			Expect(strings.Contains(err.Error(), fmt.Sprintf("invalid environment name: %s", environment.Name))).To(BeTrue())
 
-			By("Validate that Environment name starting with capital letter is not allowed.")
+		})
+
+		It("Should validate Environment CR Webhooks for invalid KubernetesClusterCredentials API URL.", func() {
+			k8sClient, err = fixture.GetE2ETestUserWorkspaceKubeClient()
+			Expect(err).To(Succeed())
+
+			if !isWebhookInstalled("environments", k8sClient) {
+				Skip("skipping as environments webhook is not installed")
+			}
+
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+			By("Validate Environment KubernetesClusterCredentials API URL while creating")
+			environment := buildEnvironmentResource("staging", "Staging Environment", "staging", appstudiosharedv1.EnvironmentType_POC)
+			environment.Spec.UnstableConfigurationFields = &appstudiosharedv1.UnstableEnvironmentConfiguration{
+				KubernetesClusterCredentials: appstudiosharedv1.KubernetesClusterCredentials{
+					APIURL: "api/test-url.com:6443",
+				},
+			}
 
-			environment.Name = "Staging"
 			err = k8s.Create(&environment, k8sClient)
-			Expect(err).NotTo(Succeed())
-			Expect(strings.Contains(err.Error(), fmt.Sprintf("Invalid value: %s", environment.Name))).To(BeTrue())
+			Expect(err).ToNot(Succeed())
+			Expect(strings.Contains(err.Error(), ": API URL must be an absolute URL starting with an 'https' scheme")).To(BeTrue())
+		})
 
-			By("Validate that Environment name having small letters is allowed.")
+		It("Should fail when updating Environment with invalid KubernetesClusterCredentials API URL", func() {
+			k8sClient, err = fixture.GetE2ETestUserWorkspaceKubeClient()
+			Expect(err).To(Succeed())
+
+			if !isWebhookInstalled("environments", k8sClient) {
+				Skip("skipping as environments webhook is not installed")
+			}
+
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+			By("Creating Environment with valid API URL")
+			environment := buildEnvironmentResource("staging", "Staging Environment", "staging", appstudiosharedv1.EnvironmentType_POC)
+			environment.Spec.UnstableConfigurationFields = &appstudiosharedv1.UnstableEnvironmentConfiguration{
+				KubernetesClusterCredentials: appstudiosharedv1.KubernetesClusterCredentials{
+					APIURL: "https://api-url.com",
+				},
+			}
 
-			environment.Name = "staging"
 			err = k8s.Create(&environment, k8sClient)
 			Expect(err).To(Succeed())
+
+			// Fetch the latest version
+			err = k8s.Get(&environment, k8sClient)
```

Comment:
I think just fetching latest copy wont work because environment controller might still processing and updating the CR which would update the CR fields like status etc.
I guess you need to wait for controller to finish updating and then fetch the latest copy and update the API URL. See this E2E for environment.
https://github.com/redhat-appstudio/managed-gitops/blob/main/tests-e2e/appstudio/environment_test.go#L82